packages/microsoft-entra-adapter/src/definitions/deploy/utils/group_deployment_life_cycle_policy.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ *                      Copyright 2024 Salto Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import _ from 'lodash'
+import { getChangeData, isModificationChange } from '@salto-io/adapter-api'
+import { GROUP_LIFE_CYCLE_POLICY_FIELD_NAME, GROUP_TYPE_NAME } from '../../../constants'
+import { validatePlainObject } from '../../type-validators'
+import { DeployRequestDefinition, DeployableRequestDefinition, EndpointPath } from '../types'
+
+export const getGroupLifecyclePolicyGroupModificationRequest = (action: 'add' | 'remove'): DeployRequestDefinition => ({
+  endpoint: {
+    path: `/groupLifecyclePolicies/{lifeCyclePolicyId}/${action}Group` as EndpointPath,
+    method: 'post',
+  },
+  transformation: {
+    pick: ['id'],
+    adjust: ({ value }) => {
+      validatePlainObject(value, GROUP_TYPE_NAME)
+      return {
+        value: {
+          groupId: value.id,
+        },
+      }
+    },
+  },
+  context: {
+    custom:
+      () =>
+      ({ change }) => {
+        const dataWithPolicyId =
+          action === 'remove' && isModificationChange(change) ? change.data.before.value : getChangeData(change).value
```

Comment:
for add action it would return undefined?

Reply:
It will return the value as it is in the 'after' value. Theoretically, it could have been just:
```suggestion
          action === 'remove' ? change.data.before.value : change.data.after.value
```

i just had to make the necessary adjustment so TS will be happy :)